comms/dht/src/discovery/service.rs

```diff
@@ -182,7 +174,22 @@ impl DhtDiscoveryService {
                     start_ts,
                 } = request;
 
-                let result = self.validate_then_add_peer(&public_key, discovery_msg).await;
+                // Make sure that the response is for the expected public key
+                if discovery_msg.public_key.as_bytes() != public_key.as_bytes() {
```

Comment:
this seems like a bannable offense

Reply:
Ah yeah this is actually checked before and banned in the dht message handler middleware, so shouldn't need to check it here again

Reply:
There are more bans, but i think lets save them for another PR